File: 1189-maximum-number-of-balloons/1189-maximum-number-of-balloons.py
class Solution:
    def maxNumberOfBalloons(self, text: str) -> int:
        freq = Counter(text)
        freq['l'] //= 2
        freq['o'] //= 2
        mini = float('inf')

        for c in "balloon":
            mini = min(mini, freq[c])

        return mini

File: 1189-maximum-number-of-balloons/1189-maximum-number-of-balloons.java
class Solution {
    public int maxNumberOfBalloons(String text) {
        int[] freq = new int[26];

        for (char c : text.toCharArray()) {
            freq[c - 'a']++;
        }

        freq['l' - 'a'] /= 2;
        freq['o' - 'a'] /= 2;
        int mini = Integer.MAX_VALUE;

        for (char c : "balon".toCharArray()){
            mini = Math.min(mini, freq[c - 'a']);
        }

        return mini;
    }
}